src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,44 @@
+package roomescape.service;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import roomescape.dao.ReservationDao;
+import roomescape.domain.Reservation;
+import roomescape.dto.ReservationListResDto;
+import roomescape.dto.ReservationReqDto;
+import roomescape.exception.ErrorCode;
+import roomescape.exception.NotFoundReservationException;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ReservationService {
+    private final ReservationDao reservationDao;
+
+    public ReservationService(ReservationDao reservationDao) {
+        this.reservationDao = reservationDao;
+    }
+    public List<ReservationListResDto> findAllReservations(){
+        List<Reservation> reservations = reservationDao.findAll();
+        return reservations.stream()
+                .map(ReservationListResDto::from)
+                .collect(Collectors.toList());
+    }
```

Comment:
```suggestion
    public List<ReservationListResDto> findAllReservations(){
        List<Reservation> reservations = reservationDao.findAll();
        return reservations.stream()
                .map(ReservationListResDto::from)
                .toList();
    }
```
- 저는 toList()가 `UnmodifiableList` 를 반환하기 때문에 주로 사용해요!
- 신희가 직접 `collect(Collectors.toList())`와 `toList()`에 대해서 학습하고 차이를 알려주세요!

Reply:
이 부분관련해서는 `toList()`가 단순히 코드를 간결하게 해주는 코드라 생각했었는데 둘이 완전히 같은 형태의 구현체로 반환되지 않는다는 점에서 차이점이 있었네요!

* `collect(Collectors.toList())` 는 ArrayList 를 반환
* `toList()` 는 `Collectors.UnmodifiableList` 또는 `Collectors.UnmodifiableRandomAccessList`를 반환

각각의 반환형태를 살펴보면 수정이 허용되는가 vs 수정이 불가능한가의 차이점으로 살펴볼 수 있을 것 같습니다. 

https://binux.tistory.com/146
https://blakes-organization.gitbook.io/java/undefined/java16/stream-list